### PR TITLE
chore: migrate to fs/promises

### PR DESCRIPTION
**Pull request by develar**
Avoid deprecation errors about using "rmdir".

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -32,7 +32,7 @@ jobs:
   test:
     parallelism: 4
     docker:
-      - image: electronuserland/builder:wine-mono
+      - image: electronuserland/builder:wine-mono-05.21
```
Comment (develar) [inline .circleci/config.yml:35]: Looks like we have a new failed test `targets` due to changed docker image :(

File: test/src/linux/linuxPackagerTest.ts [hunk 2]
```diff
@@ -177,7 +174,7 @@ test.ifNotWindows(
         await fs
           .mkdir(path.join(projectDir, "resources"), { recursive: true })
           .then(() => fs.rename(path.join(projectDir, "build", "icon.icns"), path.join(projectDir, "resources", "time.icns")))
-        await remove(path.join(projectDir, "build"))
+        await fs.rm(path.join(projectDir, "build"), { recursive: true, force: true })
```
Comment (mmaietta) [inline test/src/linux/linuxPackagerTest.ts:177]: I think this needs to be `fs.rmdir` to resolve the failing unit test

File: .circleci/config.yml (hunk 1 above)
Comment (mmaietta) [reply .circleci/config.yml:35]: >                     out={:timestamp=>"2021-05-12T09:36:17.960531+0200", :message=>"Process failed: /bin/sh failed (exit code 127). Full command was:[\"/bin/sh\", \"-c\", \"LANG=C bsdtar -czf .MTREE --format=mtree --options='!all,use-set,type,uid,gid,mode,time,size,md5,sha256,link' opt usr .PKGINFO .INSTALL\"]", :level=>:error}

`bsdtar` was removed from the base/Dockerfile in the latest PR. 

Not sure how to test that hypothesis

**Comment by mmaietta** [comment]
Hmmmm, a different 'targets' test is now failing and it isn't generating most of the expected artifacts
```
Snapshot name: `targets 1`

- Snapshot  - 16
+ Received  +  0

  Object {
    "linux": Array [
      Object {
        "arch": "x64",
-       "file": "TestApp-1.1.0.7z",
-     },
-     Object {
-       "arch": "x64",
-       "file": "TestApp-1.1.0.freebsd",
-     },
-     Object {
-       "arch": "x64",
        "file": "TestApp-1.1.0.pacman",
-     },
-     Object {
-       "arch": "x64",
-       "file": "TestApp-1.1.0.sh",
-     },
-     Object {
-       "arch": "x64",
-       "file": "TestApp-1.1.0.zip",
      },
    ],
  }
    at packAndCheck (/root/project/test/src/helpers/packTester.ts:244:27)
```

**Comment by kapalex** [comment]
Just letting you know this causes an issue with builds for node 13.x envionments (throw exception of cannot find module 'fs/promises') 
For me I ran into this when trying to build on windows 7 where node 14 is not supported.

**Comment by bromix** [comment]
Same here with Node 12.x
```
> connect@0.20.0-alpha postinstall C:\Development\ashampoo\connect\connect-launcher
> electron-builder install-app-deps

Error: Cannot find module 'fs/promises'
```
Now we are not able to create a production build for our new release :(

**Comment by Yongle-Fu** [comment]
same issue, fallback to 22.10.5 will be ok
https://github.com/electron-userland/electron-builder/issues/5953

**Comment by Yongle-Fu** [comment]
> Same here with Node 12.x
> 
> ```
> > PROJECTNAME postinstall C:\Development\ashampoo\connect\connect-launcher
> > electron-builder install-app-deps
> 
> Error: Cannot find module 'fs/promises'
> ```
> 
> Now we are not able to create a production build for our new release :(

Same here with Node 12.x, fallback to 22.10.5 will be ok
#5953

**Comment by alectrocute** [comment]
This broke our CI builds today.

**Comment by nuochong** [comment]
me too

**Pull request closed without merging**